pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -171,16 +171,20 @@ func (c *Reconciler) ensureSinkIsPublisher(ctx context.Context, s *v1.CloudAudit
 // previously created stackdriver sink.
 func (c *Reconciler) deleteSink(ctx context.Context, s *v1.CloudAuditLogsSource) error {
 	if s.Status.StackdriverSink == "" {
+		s.Status.MarkSinkNotReady("SinkFinalized", "No sink to delete.")
 		return nil
 	}
 	logadminClient, err := c.logadminClientProvider(ctx, s.Status.ProjectID)
 	if err != nil {
 		logging.FromContext(ctx).Desugar().Error("Failed to create LogAdmin client", zap.Error(err))
+		s.Status.MarkSinkNotReady(deleteSinkFailed, "Failed to create LogAdmin Client: %s", err.Error())
 		return err
 	}
-	if err = logadminClient.DeleteSink(ctx, s.Status.StackdriverSink); status.Code(err) != codes.NotFound {
+	if err = logadminClient.DeleteSink(ctx, s.Status.StackdriverSink); err != nil && status.Code(err) != codes.NotFound {
+		s.Status.MarkSinkNotReady(deleteSinkFailed, "Failed to delete Stackdriver sink: %s", err.Error())
 		return err
 	}
+	s.Status.MarkSinkNotReady("SinkDeleted", "Sink successfully deleted: %s", s.Status.StackdriverSink)
```

Comment:
I'm not entirely sure we should be setting this in the success case, as we will try to update the status of a deleted resource, which will probably give some error...
can you confirm in the controller logs that this doesn't give an error?

Reply:
if we decide not to, we should remove it in other places as well

Reply:
I think we should make the behavior consistent.
When @grac3gao marked the sub-conditions in finalizer of broker, https://github.com/google/knative-gcp/pull/1295/files, we won't mark as Failed or NotReady when we successfully delete the resource.
If broker and pubSub finalizer are not consistent, we should try to reach an agreement and make it consistent among all resources.

Reply:
@nachocano 
I checked this issue https://github.com/knative/eventing/issues/3305
It looks like `it was confirmed that we can update the status of a resource that is being deleted.`
@grac3gao  Can you help confirm that we can update the status of a resource that is being deleted?

Reply:
Yes, we are doing so when it fails to delete. But I wasn't sure whether we should also do it when the deletion succeeds. Might not hurt, but I thought it was not needed.

Reply:
@nachocano I agree it is not needed to mark the condition when the deletion succeeds if both deletion and creation are using the same condition type. Marking it failed when the deletion succeeds is kind of confusing. 
My point is that we should make the mark behavior consistent across different resources.

Reply:
In Broker, we didn't mark the condition if the deletion succeeds (we only mark the condition to provide information if the deletion failed for some reason, and it is confirmed that we can update the condition in this case). IMO, If the deletion succeeds, then the resource will just removed from the namespace, user doesn't need (and they are not interested) to know the condition. Only deletion failure will cause the resource hang in the namespace, and we need to let the resource communicate to user when the users are looking at object's condition.

Reply:
@nachocano I checked the controller logs that it doesn't gave an error regarding the finalizer. @capri-xiyue Yes, the PubSub deletion also marks Pubsub Fail when the deletion succeed. We should keep everything consistent.
I agree with all of your idea that marking resources NotReady or Fail is not needed when the deletion is successful, so I will remove it here. I will create a separate issue to make PubSub consistent with this logic in a separate PR. Is that ok with all of you?

Reply:
SGTM

Reply:
sg